design-proposals/shadow-node.md

```diff
@@ -0,0 +1,328 @@
+- These steps can be broken to multiple seemingly unrelated PRs, that are easier to review.
+
+Cons:
+- Relying on a third part tools.
+- Using third-party tools cannot filter only the kubevirt.io related labels.
+- annotations are not addressed (for example `kubevirt.io/ksm-handler-managed` annotation)
+- with NFD one cannot filter only kubevirt owned labels, making it a bit less safe.
+
+## Solution option #5: [Ruled-out] Using status section
+Same proxy solution as the primary solution where virt-hadler updates the labels/annotations somewhere, and it gets copied to the node by virt-controller, but instead of using a new CRD as the medium - writing the labels/Annotations to the status section of an existing CRD like kubevirt CR.
+
+Pros:
+- Avoid introducing a new CRD to the project.
+
+Cons:
+- [Deal-breaker] The status section is considered a user facing API section. Dynamically adding/updating 150+ technical labels there may only confuse the user and thus is not advisable.
+
+## Solution option #6: [Ruled-out] Using the virt-handler pod as a medium to pass the labels/annotations
+Same proxy solution as primary solution where virt-handler updates the labels/annotations somewhere, and it gets copied to the node by virt-controller, but instead of using a new CRD as the medium - writing the labels/annotations to the virt-handler pods's labels/annotations.
+Labels/Annotations that need to pass to the node should get a special prefix (like "nodesync/"), so that regular labels don't get passed to the node by mistake.
+
+Pros:
+- Avoid introducing a new CRD to the project.
+- A simple and clean approach.
+
+Cons:
+- [Deal-breaker] It requires giving the virt-handler RBAC permission to update pods, which is generally considered risky.
+- It's only relevant for passing labels/annotations. CRD approach allows for future expansions.
+
+## Solution option #7: [Ruled-out] Fine-tune each virt-handler RBAC to only patch its own node.
+Introducing a new controller called token-distributor-controller. It will create a token for each virt-handler pod and deliver it to the node.
+
+Pros:
+- Avoid introducing a new CRD to the project.
+- A malicious user can't manipulate pods from other nodes to migrate to their node.
+
+Cons:
+- A malicious user can still manipulate its own node with non-kubevirt controlled params/labels, then possibly lure control plane components to its node in order to steal their credentials. For example, it can add the control-plane label to its own node, then add other labels to attract components that seek specific functionalities, and take over them.
+
+This solution is not a bad one, but it's more important to address not being able to affect non-kubevirt labels/annotations before doing this (see Follow-up Steps section).
+
+# chosen solution
+
+TBD until further discussion.
+
+# Update/Rollback Compatibility
```

Comment:
TBD per chosen solution

Reply:
Should I delete the stuff currently written?